src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIBuildJobExecutionService.java

```diff
@@ -460,4 +498,29 @@ private LocalCIBuildResult constructBuildResult(List<LocalCIBuildResult.LocalCIT
 
         return new LocalCIBuildResult(assignmentRepoBranchName, assignmentRepoCommitHash, testsRepoCommitHash, isBuildSuccessful, buildRunDate, List.of(job));
     }
+
+    private Path cloneAndCheckoutRepository(ProgrammingExerciseParticipation participation, String commitHash) {
+        try {
+            // Clone the assignment repository into a temporary directory with the name of the commit hash and then checkout the commit hash.
+            Repository repository = gitService.getOrCheckoutRepository(participation.getVcsRepositoryUrl(), Paths.get("checked-out-repos", commitHash), false);
+            gitService.checkoutRepositoryAtCommit(repository, commitHash);
+            return repository.getLocalPath();
+        }
+        catch (GitAPIException e) {
+            throw new LocalCIException("Error while cloning repository", e);
+        }
+    }
+
+    private void deleteCloneRepo(ProgrammingExerciseParticipation participation, String commitHash) {
+        try {
+            Repository repository = gitService.getOrCheckoutRepository(participation.getVcsRepositoryUrl(), Paths.get("checked-out-repos", commitHash), false);
```

Comment:
reads a bit weird that in order to delete the repo it's in theory cloned first. Is there only a `get` function as well?

Reply:
Yes, I get your point. Found `getExistingCheckedOutRepositoryByLocalPath()` and used it instead